desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -115,11 +116,33 @@ def handle_on_link_shared(channel_id, message_ts, links):
       msg = "Document with {key}={value} does not exist".format(key='uuid' if id_type == 'uuid' else 'id', value=qid)
       raise PopupException(_(msg))
 
-    payload = _make_unfurl_payload(item['url'], id_type, doc, doc_type)
-    response = slack_client.chat_unfurl(channel=channel_id, ts=message_ts, unfurls=payload)
+    # Mock request for query execution and fetch result
+    user = rewrite_user(User.objects.get(username=doc.owner.username))
+    request = MockRequest(user=user)
+
+    payload = _make_unfurl_payload(request, item['url'], id_type, doc, doc_type)
+    response = slack_client.chat_unfurl(channel=channel_id, ts=message_ts, unfurls=payload['payload'])
     if not response['ok']:
       raise PopupException(_("Cannot unfurl link"), detail=response["error"])
-
+    
+    # Generate and upload result xlsx file only if result available
+    if payload['file_status']:
+      notebook = json.loads(doc.data)
+      snippet = notebook['snippets'][0]
+      snippet['statement'] = notebook['snippets'][0]['statement_raw']
+      file_format = 'xlsx'
+      file_name = _get_snippet_name(notebook)
+
+      content_generator = get_api(request, snippet).download(notebook, snippet, file_format='xls')
```

Comment:
Tried it but when giving `file_format = 'xls'` in `file_upload` it upload the result as a `zip file`
<img width="413" alt="Screenshot 2021-03-21 at 11 58 30 AM" src="https://user-images.githubusercontent.com/42064744/111896034-c30cd100-8a3c-11eb-89d4-eb4288960df7.png">
 
However when this file is downloaded and opened, it opens as `xls` successfully. So, it does the job but does not looks good seen as a zip file @romainr 
So, I thought its better to write 'xls' once instead of writing 'xlsx' twice in file_upload or initializing a new var for it.